### ISSUE TITLE
Compiler crash when overriding method that returns a generic pointer

### ISSUE THREAD

**Issue by TChatzigiannakis** [dotnet/roslyn#37051]
**Version Used**: 

3.100.19.26603 (9d80dea7)

**Steps to Reproduce**:

1. Create a base class with an abstract method that returns a `T*` (of course `where T : unmanaged`).
2. Create a derived class that overrides that method.

Or you can just compile a .cs file with this code:

```cs
using System;

public unsafe abstract class Base
{
    public abstract T* Alloc<T>()
        where T : unmanaged;
}

public unsafe class Derived : Base
{
    public override T* Alloc<T>()
    {
        throw new NotImplementedException();
    }
}
```

**Expected Behavior**:

The compiler should accept this program (if it is correct) or print a diagnostic (if it is incorrect).

**Actual Behavior**:

The compiler crashes. When used from Visual Studio, the IDE abruptly shuts down. When used from VS Code, the editor shows the text `!!MISSING: command!!` above the offending methods.

**Comment by RikkiGibson** [comment]
Quick and easy way to find out the compiler version being used is to add `#error version` into one of your C# files

**Comment by RikkiGibson** [comment]
It seems like the MethodSymbol we produce for `Derived.Alloc<T>` is malformed somehow. A lot of its members throw NRE (such as ToString()...)

**Comment by RikkiGibson** [comment]
This appears to also be the behavior in dev15: https://sharplab.io/#v2:C4LglgNgNAJiDUAfAAgJgIwFgBQyDMABAK4B2AzgIYBmApgRQEZnABOFAxsAWgQEIVkaOAN44C47oUbM2nAgBUAVAQCCECAHt2AHnkA+ABQBKMRLMB3ABY0WdeQRDESAWwokKAcxowA3DgC+OPhOlLTcqAQAIjZgAG7eDnwCQtii2GbBGvEsLGAwdspqmjr6xqbiaWZmwJYsGuYEJDQNAHIawACSzgAOEDTONCTA3gCiAB7sNN3AYBokxn7pEoHY/kA=

**Comment by RikkiGibson** [comment]
Fixed by #37320. Thanks for bringing the issue to our attention @TChatzigiannakis!
